addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,375 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+import json
+import logging
+import requests
+from werkzeug.urls import url_quote
+from base64 import b64encode
+
+from odoo import api, models, _
+from odoo.tools.float_utils import float_repr, float_round
+
+_logger = logging.getLogger(__name__)
+
+
+ETA_DOMAINS = {
+    'preproduction': 'https://api.preprod.invoicing.eta.gov.eg',
+    'production': 'https://api.invoicing.eta.gov.eg',
+    'token.preproduction': 'https://id.preprod.eta.gov.eg',
+    'token.production': 'https://id.eta.gov.eg',
+}
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    @api.model
+    def _l10n_eg_get_eta_api_domain(self, production_enviroment=False):
+        return production_enviroment and ETA_DOMAINS['production'] or ETA_DOMAINS['preproduction']
+
+    @api.model
+    def _l10n_eg_get_eta_token_domain(self, production_enviroment=False):
+        return production_enviroment and ETA_DOMAINS['token.production'] or ETA_DOMAINS['token.preproduction']
+
+    @api.model
+    def _l10n_eg_eta_connect_to_server(self, request_data, request_url, method, access_token=False, production_enviroment=False):
+        api_domain = access_token and self._l10n_eg_get_eta_token_domain(production_enviroment) or self._l10n_eg_get_eta_api_domain(production_enviroment)
```

Comment:
I would think that if you have an access_token, you should get the api domain...  @akr-odoo